### PR TITLE
fix: aa antipaterns

### PR DESCRIPTION
**Pull request by nksazonov**

### CODE REVIEW

**Review by philanton — APPROVED** [review]

File: pkg/universal_sigver/client.go [hunk 1]
```diff
@@ -47,24 +47,24 @@ type Client interface {
 type backend struct {
 	provider          *ethclient.Client
 	smartWalletConfig *smart_wallet.Config
-	entryPointAddress *common.Address
+	entryPointAddress common.Address
 }
 
-func NewUniversalSigVer(providerURL string, smartWalletConfig *smart_wallet.Config, entryPointAddress *common.Address) Client {
+func NewUniversalSigVer(providerURL string, smartWalletConfig *smart_wallet.Config, entryPointAddress common.Address) (Client, error) {
```
Comment (MaxMoskalenko) [inline pkg/universal_sigver/client.go:53]: What's the reason of using config as a pointer?

Comment (nksazonov) [reply pkg/universal_sigver/client.go:53]: So that it is not copied to memory.

Comment (MaxMoskalenko) [reply pkg/universal_sigver/client.go:53]: It is a good practice to not use pointers, where they are not supposed to be

Comment (nksazonov) [reply pkg/universal_sigver/client.go:53]: Understood, thanks

Comment (nksazonov) [reply pkg/universal_sigver/client.go]: [1a7f206](https://github.com/layer-3/clearsync/pull/270/commits/1a7f20623aea652ebaf1504d4c9f7b53da5bb84e)

**Pull request merged**